woocommerce/woocommerce-admin
# Fix report arguments handling in CSV export.

Fixes #5503.

This PR fixes the CSV Report Export handling of report arguments/parameters. Before this change, the arguments were not mapped (in Controller) and were not sanitized (using the schema defined in the `Controller::get_collection_params`).

In writing the test case, I found a bug with the handling of empty category results for products (the `-1` placeholder ID used to facilitate the query was showing up in results) - this PR includes a fix for that as well.

### Detailed test instructions:

- Go to Analytics > Categories
- Choose a single category to filter
- If necessary, edit the rows per page to span several pages to trigger an email rather than a direct download
- Notice that the emailed report **does not include** additional products not in the selected category.
